### ISSUE TITLE
Invoke-OAIChat only echoes prompt

### ISSUE THREAD

**Issue by ToddKlindt** [dfinke/PSAI#38]
I'm trying to play with Invoke-OAIChat and I've got something set up wrong. When I send it a prompt, all I get back is the prompt. It looks like this:
![image](https://github.com/user-attachments/assets/3a0bb909-1746-40fa-a9a8-f3b2d5ddf1ca)
I get this on both machines I've tried it on. I know my keys are okay because if I put in a known bad key I get an error. I am trying both a Project API key and a user API key. Both give me the same experience. I have tried the -verbose switch. I get a lot of output, but nothing that obviously points to what's going wrong.

**Comment by jmcanally** [comment]
Same experience for me

**Comment by jmcanally-tricentis** [comment]
I resolved my issue, I was out of credits with OpenAI. 😑

**Comment by jmcanally** [comment]
Resolved my issue, I was out of credits on OpenAI. 😑

**Comment by ToddKlindt** [comment]
Good idea to check that. I think I'm okay as I have set a $50 limit. When I look at the usage this is what I see:
![image](https://github.com/user-attachments/assets/45cc8c97-8020-49c2-be7b-78c15a23a5bf)

**Comment by ToddKlindt** [comment]
I fixed it. I had to add credit to my OpenAI account. I thought it would pull it as it needed it. I added $10 and now it all works. 

Thanks @jmcanally for the idea.

**Comment by dfinke** [comment]
Thanks @ToddKlindt for trying out the module and reporting.

Thanks @jmcanally for responding.

Yes, when OpenAI moved to the credit approach, the REST calls in the PowerShell do not get an error on the initial setup and just returns an empty reponse.

Glad you figured it out.

**Comment by ToddKlindt** [comment]
> Thanks @ToddKlindt for trying out the module and reporting.
> 
> Thanks @jmcanally for responding.
> 
> Yes, when OpenAI moved to the credit approach, the REST calls in the PowerShell do not get an error on the initial setup and just returns an empty reponse.
> 
> Glad you figured it out.

Is there a way for you to check for an empty response and give the user some feedback about what might be the issue? I initially tried this module a couple of months ago and gave up on it because I couldn't get it to work. 

BTW, I heard you on RunAsRadio this week. Good interview. It's what encouraged me to try this again. I'm doing a talk this week at a community event on writing PowerShell with AI. I'll be sure to give this a plug.

**Comment by dfinke** [comment]
@ToddKlindt thanks!

Doing a check does not seem like the best answer. I also use this in the Agent components.

Unfortunately OpenAI is still "preview" I would expect an error return code which is what the module checks for.

Unfortunately, very difficult to figure this out. Seems like it only happens for new accounts.
